torch/csrc/jit/frontend/sugared_value.h

```diff
@@ -651,5 +651,44 @@ struct SimpleSelf : public Self {
  private:
   ClassTypePtr classType_;
 };
+
+
+// This is not a SimpleValue so it can pass through the code paths
+// that expect a SimpleValue as a sugared value
+struct TORCH_API ExceptionMessageValue : public SugaredValue {
+  ExceptionMessageValue(Value* value) : value_(value) {}
+
+  std::string kind() const override {
+    return "exception message";
+  }
+
+  Value* getValue() {
+    return value_;
+  }
+
+  Value* value_;
+};
+
+
+struct TORCH_API ExceptionValue : public SugaredValue {
+  ExceptionValue(const std::string& message) : message_(std::move(message)) {}
```

Comment:
Do we need both `ExceptionValue` and `ExceptionMessageValue` ? or could one extend from the other ?

Reply:
`ExceptionValue` corresponds to `RuntimeError` and `ExceptionMessageValue` corresponds to the instance returned by `RuntimeError("error")`, so we do need both since it is valid to do something like `raise RuntimeError` and `raise RuntimeError("error")`